data/recommedation_algo/recommender.py

```diff
@@ -129,23 +124,47 @@ def _generate_similar_movies(self, user_list):
 
         return similar_list
 
-    def _generate_recommend_seeds(self, user_pool):
-        """
-        given user watching history, generate similar
-        movies that may be recommended
-        :param user_pool: list
-        :return: list
-        """
-        user_list = []
-        for user_rating in user_pool:
-            movie_id, score = user_rating
-            if score >= self.USER_RATINGS_CRITERION:  # consider as favorable movie
-                user_list.append(movie_id)
-        return user_list
+    def _generate_recommend_list(self, similar_list, user_id):
+        logging.debug("initialising predicting process ...")
+
+        scale = UserScale(user_id)
+
+        recommend_list = []
+
+        logging.debug("size of similar list: " + str(len(similar_list)))
+        for potential in similar_list:
+            public_ratings = self.db.get_public_rating_dict(potential)
+
+            if len(public_ratings) == 0:
+                logging.debug("rating does not exist, extracting ratings ...")
+                self.controller.update_single_movie_rating(potential)
+                public_ratings = self.db.get_public_rating_dict(potential)
+
+            if len(public_ratings) == 3:
```

Comment:
3 is temporary value for now since only 3 sources, in the future can be len(public ratings) == sizeof (rating source)